### PR TITLE
Added '--config' flag support for both sspl_test and sspl_reset scripts

### PR DESCRIPTION
**Pull request by gargpalak048**
## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
  </code>
</pre>
## Need to add --config flag for sspl_test & sspl_reset scripts
<pre>
  <code>
    Please add Problem decription here...
  </code>
</pre>
## Solution
<pre>
  <code>
    Added --config flag for future use-case
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 1]
```diff
@@ -252,27 +252,19 @@ def __init__(self, args):
 
     def validate(self):
         if self.args:
-            i=0
-            while i < len(self.args):
-                # Provision for --config <global_config_url> for future use-case.
-                if self.args[i] == "--config":
-                    try:
+            try:
+                for i in range(len(self.args)):
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:256]: read args in one for loop

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py [hunk 2]
```diff
@@ -27,8 +27,15 @@ def __init__(self, args: list):
         self.name = "sspl_test"
 
     def process(self):
-        # TODO: Need to convert run_tests.sh from shell to python.
-        CMD = f"{TEST_DIR}/run_tests.sh test {' '.join(self.args)}"
+        # TODO: Need to convert run_tests.sh from shell to python
+        test_plan=None
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_test.py:31]: this script too can take --plan argument and ignore anything else coming.

Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py:31]: @cdeshmukh sspl_test.py will accept all the args from sspl_setup, but will pass only test plan arg to the run_test.sh script will ignore other args.
Do we need any other code flow?

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py (hunk 1 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_setup.py]: Agree.Done the changes.

File: low-level/files/opt/seagate/sspl/setup/sspl_test.py (hunk 2 above)
Comment (cdeshmukh) [reply low-level/files/opt/seagate/sspl/setup/sspl_test.py:31]: i understand the flow you put in place. Am saying though parsing should be generic and user friendly, which existing script is not. So adding --plan here too would help.

Its not functional part am talking about, but code maintainability & user friendliness

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def usage(prog: str):
             [ init --config [<global_config_url>] ]
             [ config --config [<global_config_url>] ]
             [ test --config [<global_config_url>] --plan [sanity|alerts|self_primary|self_secondary] ]
-            [ reset [hard|soft] ]
+            [ reset --config [<global_config_url>] [hard|soft] ]
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:59]: add --type

**Review by cdeshmukh — APPROVED** [review]

**Review by saurabhkhanvilkar — APPROVED** [review]

**Review by thavanathan — APPROVED** [review]

**Pull request merged**